securedrop/journalist_app/api2/shared.py

```diff
@@ -0,0 +1,50 @@
+"""
+This module contains helper functions factored out of the v1 Journalist API
+(journalist_app.api) and shared with the v2 Journalist API (journalist_app.api2)
+"""
+
+from os import path
+from uuid import UUID
+
+from db import db
+from journalist_app.sessions import session
+from models import (
+    Reply,
+    SeenReply,
+    Source,
+)
+from sqlalchemy.exc import IntegrityError
+from store import Storage
+
+
+def save_reply(source: Source, data: dict) -> Reply:
+    source.interaction_count += 1
+    filename = Storage.get_default().save_pre_encrypted_reply(
+        source.filesystem_id,
+        source.interaction_count,
+        source.journalist_filename,
+        data["reply"],
+    )
+
+    # issue #3918
```

Comment:
nit: Let's make this more useful instead of just pointing to https://github.com/freedomofpress/securedrop/issues/3918 since we're modifying it...
```suggestion
    # We only save the stored reply's basename and not the whole storage path
```